src/main/java/games/strategy/triplea/ui/menubar/LobbyMenu.java

```diff
@@ -160,13 +167,15 @@ private void addBanUsernameMenu(final JMenu parentMenu) {
           return;
         }
       }
-      final Instant expire = Instant.now().plusMillis(requestTimespanSupplication());
+      final Optional<TemporalAmount> duration = promptBanDuration();
       final IModeratorController controller = (IModeratorController) lobbyFrame.getLobbyClient().getMessengers()
           .getRemoteMessenger().getRemote(ModeratorController.getModeratorControllerName());
-      try {
-        controller.banUsername(new Node(name1, InetAddress.getByName("0.0.0.0"), 0), Date.from(expire));
-      } catch (final UnknownHostException ex) {
-        ClientLogger.logQuietly(ex);
+      if (duration.isPresent()) {
```

Comment:
Using `Optional#ifPresent()` here would be more idiomatic Java 8.

Reply:
Good Point, allows us to save a line as well...